.repo/sample-list-generator/src/index.ts

```diff
@@ -0,0 +1,75 @@
+import path from 'path';
+import fs from 'fs/promises';
+
+import { AVAILABLE_FOLDERS, REPO_BASE_URL } from './constants';
+import { getApiListForSample } from './api-detector';
+import type { IAvailableFolderItem, ISampleItem } from './types';
+import { getBasePath, isFileExists } from './utils';
+
+const getAllSamples = async () => {
```

Comment:
Please move sample collection into a separate class, index.ts should only orchestrate and not include any actual logic.